src/classes/HH_Container_LCTRL.cls

```diff
@@ -80,31 +83,55 @@ public class HH_Container_LCTRL {
         try {
             if (hhId == null)
                 return null;
-            string strSoql;
-            if (UTIL_Describe.isObjectIdThisType(hhId, 'Account')) {
-                strSoql = 'select Id, Name, npo02__SYSTEM_CUSTOM_NAMING__c, npo02__Formal_Greeting__c, npo02__Informal_Greeting__c, ' +
-                        UTIL_Namespace.StrTokenNSPrefix('Number_of_Household_Members__c') + ', ' +
-                        'BillingStreet, BillingCity, BillingState, BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude';
-
-                /*** until we support these, we don't want them to interfere with what the user specifies.
-                if (ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled)
-                    strSoql += ', BillingCountryCode, BillingStateCode ';
-                ****/
-
-                strSoql += ' from Account where Id = :hhId';
-            } else {
-                strSoql = 'select Id, Name, npo02__SYSTEM_CUSTOM_NAMING__c, npo02__Formal_Greeting__c, npo02__Informal_Greeting__c, ' +
-                        UTIL_Namespace.StrTokenNSPrefix('Number_of_Household_Members__c') + ', ' +
-                        'npo02__MailingStreet__c, npo02__MailingCity__c, npo02__MailingState__c, npo02__MailingPostalCode__c, npo02__MailingCountry__c ' +
-                        ' from npo02__Household__c where Id = :hhId';
+            
+            string strSoql = 'select Id, ';
+            String strComma = '';
+            Boolean isHHAccount = UTIL_Describe.isObjectIdThisType(hhId, 'Account');
+            list<String> listStrFld = getUpdateableHouseholdFields(isHHAccount);
+            String strObject = isHHAccount ? 'Account' : 'npo02__Household__c';
+            
+			// check field level security and construct our soql            
+            for (String strF : listStrFld) {
+                DescribeFieldResult dfr = UTIL_Describe.getFieldDescribe(strObject, strF);
+                if (dfr == null || !dfr.isAccessible() || !dfr.isUpdateable())
+                    throw (new permsException(string.format(label.flsError, new string[]{strObject + '.' + strF})));
+                
+                strSoql += strComma + strF;
+                strComma = ',';
             }
+            
+            strSoql += ' from ' + strObject + ' where Id = :hhId';
```

Comment:
Do you also need to add a CRUD check on the object?  Or is checking the fields enough?